keps/sig-network/4444-service-routing-preference/README.md

```diff
@@ -0,0 +1,945 @@
+      parsed the `routingPreference` value. A `False` status typically suggests
+      a configuration issue (e.g., an unsupported or malformed preference).
+    * The EndpointSlice controller will set this status to `True` if it
+      recognizes the `routingPreference` value as one it explicitly supports. 
+
+* RoutingPreferenceProgrammed
+  * **Type:**  `RoutingPreferenceProgrammed`
+  * **Description:** 
+    * This condition indicates whether they `routingPreference` has generated
+      some configuration that is assumed to be ready soon in the underlying data
+      plane.
+    * The EndpointSlice controller will set this status to `True` if it
+      successfully populated the EndpointSlice hints based on the
+      `routingPreference`.
+
+Note that the EndpointSlice controller and kube-proxy implementation does not
+currently provide a condition denoting acknowledgment from the dataplane. In the
+future when this is possible, another condition like the following could be
+used:
+
+* RoutingPreferenceReady
+  * **Type:** `RoutingPreferenceReady`
+  * **Description:** Confirms that the dataplane has received the hints,
+    acknowledged them, and configured itself accordingly.
+
+#### Condition usage by other implementations
+
+Other implementations supporting `routingPreference` **should** adopt a domain
+prefixing strategy for their condition types. This means prefixing condition
+types with a domain string (e.g., `my.domain.io/RoutingPreferenceAccepted`) to
+prevent conflicts when multiple control planes (like the default EndpointSlice
+controller) are present.
+
+<<[UNRESOLVED Name for the field is being discussed]>>
+
+### Choice of field name
+The name `routingPreference` is meant to capture the highly
+implementation-specific nature of this field and how it affects the routing of
+traffic
+* Field names that include the word "topology", like `topologyPreference` and
+  `topologyRoutingPreference`, were avoided because:
+  * Topology word might be a bit vague
+  * The actual heuristic might not specifically be tied to the geographical or
+    network topology, but might also be considering factors like latency, active
+    connections, cpu usage of the backends, etc.
+* Use of the word “hint” for names like `trafficRoutingHint` and
+  `customRoutingHint` was avoided so as not to confuse with the hint field
+  within EndpointSlices which has previously been discussed in relation to such
+  routing heuristics. In a sense, the hint field within EndpointSlices is just
+  an implementation detail.
+* Use of words like “traffic” and “policy” for names like
+  trafficRoutingPreference was not favored so that it’s clearly different from
+  the `internalTrafficPolicy` and `externalTrafficPolicy` fields which have a
+  more fixed behavior.
+* Use of the word "selection" for a name like `endpointSelection` were avoided
+  so as not to confuse with the actual process of selecting the complete set of
+  pods backing a service.
```

Comment:
`endpointSelection` was my suggestion before I got to this comment... I'm not sure if I agree or disagree with that objection...

maybe `endpointDistribution`?